### PR TITLE
Add Aspire Integrations

### PR DESCRIPTION
**Pull request by jcocchi**
Add Aspire integrations for
- Azure Cosmos DB
- Azure OpenAI

### CODE REVIEW

File: src/cosmos-copilot.WebApp/Program.cs [hunk 1]
```diff
@@ -1,12 +1,55 @@
...
 
+// Add service defaults & Aspire components.
+builder.AddServiceDefaults();
+
 builder.RegisterConfiguration();
 builder.Services.AddRazorPages();
 builder.Services.AddServerSideBlazor();
+
+// Configure Azure Cosmos DB Aspire integration
+builder.AddAzureCosmosClient(
```
Comment (markjbrown) [inline src/cosmos-copilot.WebApp/Program.cs:17]: add this line before adding aspire cosmos client.

var cosmosEndpoint = builder.Configuration.GetSection(nameof(CosmosDb)).GetValue<string>("Endpoint");

File: src/cosmos-copilot.WebApp/Program.cs [hunk 2]
```diff
@@ -1,12 +1,55 @@
...
+    "cosmos",
+    settings =>
+    {
+        settings.Credential = new DefaultAzureCredential();
+        settings.DisableTracing = false;
+    },
```
Comment (markjbrown) [inline src/cosmos-copilot.WebApp/Program.cs:23]: add this line to settings 

settings.AccountEndpoint = new Uri(cosmosEndpoint!);

File: src/cosmos-copilot.WebApp/Program.cs [hunk 3]
```diff
@@ -45,41 +88,8 @@ public static void RegisterConfiguration(this WebApplicationBuilder builder)
...
-            {
-                return new OpenAiService(
-                    endpoint: openAiOptions.Value?.Endpoint ?? String.Empty,
-                    completionDeploymentName: openAiOptions.Value?.CompletionDeploymentName ?? String.Empty,
-                    embeddingDeploymentName: openAiOptions.Value?.EmbeddingDeploymentName ?? String.Empty
-                );
-            }
-        });
+        services.AddSingleton<CosmosDbService, CosmosDbService>();
+        services.AddSingleton<OpenAiService, OpenAiService>();
```
Comment (markjbrown) [inline src/cosmos-copilot.WebApp/Program.cs:92]: Change this line
 var semanticKernalOptions = provider.GetRequiredService<IOptions<SemanticKernel>>();

to this.
var semanticKernalOptions = provider.GetRequiredService<IOptions<OpenAi>>();

can also get rid loading the configuration for SemanticKernel completely in RegisterConfiguration or maybe keep around and move configurations from Chat into it. That might make more sense depending on what Aayush does with the Semantic Kernel service class.

File: src/cosmos-copilot.WebApp/Properties/serviceDependencies.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "dependencies": {
+    "cosmosdb1": {
```
Comment (markjbrown) [inline src/cosmos-copilot.WebApp/Properties/serviceDependencies.json:3]: This doesn't seem to be use anywhere :)

**Comment by markjbrown** [comment]
See comments above on what I changed to get it to start.

One thing I realize. I don't think the app is using Aspire anywhere when it executes. Wouldn't the CosmosDB, OpenAI and Semantic Kernel classes need to instantiate the Aspire components? It doesn't appear to.

**Comment by jcocchi** [comment]
> `One thing I realize. I don't think the app is using Aspire anywhere when it executes. Wouldn't the CosmosDB, OpenAI and Semantic Kernel classes need to instantiate the Aspire components? It doesn't appear to.`

Aspire is what allows the clients to be injected directly into the service as opposed to being instantiated in the service constructor like they used to be. The Aspire components for CosmosDB and OpenAI are in Program.cs of the web app, the Semantic Kernel service doesn't use Aspire yet

**Pull request merged**